src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,171 @@
+                char digraphchar1{modifiedinput[k]};
+                char digraphchar2{modifiedinput[k+1]};
+
+                // - Find the coords in the grid for each digraph
+
+                auto iterindex1{playfairciphermap_.find(digraphchar1)};
+                auto iterindex2{playfairciphermap_.find(digraphchar2)};
+                std::pair<int,int> charcoord1{(*iterindex1).second};
+                std::pair<int,int> charcoord2{(*iterindex2).second};
+
+                std::cout << "coordinate char 1: " << charcoord1.first << ":" << charcoord1.second << std::endl;
+                std::cout << "coordinate char 2: " << charcoord2.first << ":" << charcoord2.second << std::endl;
+
+                // - Apply the rules to these coords to get 'new' coords
+
+                if(charcoord1.first == charcoord2.first){
+                    //same row rule
+                    std::cout << "row rule" << std::endl;
+                    if(charcoord1.second == 5){
+                        charcoord1.second = 1;
+                    } else {
+                        charcoord1.second++;
+                    }
+
+                    if(charcoord2.second == 5){
+                        charcoord2.second = 1;
+                    } else {
+                        charcoord2.second++;
+                    }
+                } else if (charcoord1.second == charcoord2.second) {
+                    //same column rule
+                    std::cout << "column rule" << std::endl;
+                    if(charcoord1.first == 5){
+                        charcoord1.first = 1;
+                    } else {
+                        charcoord1.first++;
+                    }
+
+                    if(charcoord2.first == 5){
+                        charcoord2.first = 1;
+                    } else {
+                        charcoord2.first++;
+                    }
+                } else {
+                    //rectangle rule
+                    std::cout << "rectangle rule" << std::endl;
+                    int temp{charcoord1.second};
+                    charcoord1.second = charcoord2.second;
+                    charcoord2.second = temp;
+                }
+
+                // - Find the letter associated with the new coords
+
+                auto iterinvindex1{playfairciphermapinv_.find(charcoord1)};
+                auto iterinvindex2{playfairciphermapinv_.find(charcoord2)};
+                char newchar1{(*iterinvindex1).second};
+                char newchar2{(*iterinvindex2).second};
+
+                std::cout << "new coord 1: " << charcoord1.first << ":" << charcoord1.second << std::endl;
+                std::cout << "new coord 2: " << charcoord2.first << ":" << charcoord2.second << std::endl;
+
+                output += newchar1;
+                output += newchar2;
+
+                std::cout << "new characters: " << newchar1 << " " << newchar2 << std::endl;
+            }
+            break;
+        case CipherMode::Decrypt:
```

Comment:
This is actually just as above only the shift is reversed (`-1` rather than `+1) . You can use the same code but have a variable representing the shift that is changed depending on if encrypt or decrypt is required.